config/ompi_check_pvfs2.m4

```diff
@@ -46,70 +46,27 @@ AC_DEFUN([OMPI_CHECK_PVFS2],[
              [Build Pvfs2 support, optionally adding DIR/include, DIR/lib, and DIR/lib64 to the search path for headers and libraries])])
     OPAL_CHECK_WITHDIR([pvfs2], [$with_pvfs2], [include/pvfs2.h])
 
-    AC_ARG_WITH([pvfs2-libs],
-        [AC_HELP_STRING([--with-pvfs2-libs=LIBS],
-                       [Libraries to link with for pvfs2])])
-
-    temp_with_pvfs2="$with_pvfs2"
     AS_IF([test -z "$with_pvfs2"],
-          [with_pvfs2="/usr/local"])
+          [ompi_check_pvfs2_dir="/usr/local"],
+          [ompi_check_pvfs2_dir="$with_pvfs2"])
 
-    temp_with_pvfs2_libs="$with_pvfs2_libs"
-    AS_IF([test -z "$with_pvfs2_libs"],
-	[with_pvfs2_libs="pvfs2 pthread"])
+    if test -e "$ompi_check_pvfs2_dir/lib64" ; then
+        ompi_check_pvfs2_libdir="$ompi_check_pvfs2_dir/lib64"
+    else
+        ompi_check_pvfs2_libdir="$ompi_check_pvfs2_dir/lib"
+    fi
 
     # Add correct -I and -L flags
-    AS_IF([test -d "$with_pvfs2/include"],
-        [check_pvfs2_CPPFLAGS="-I$with_pvfs2/include"
-            $1_CPPFLAGS="$check_pvfs2_CPPFLAGS"
-            CPPFLAGS="$CPPFLAGS $check_pvfs2_CPPFLAGS"],
-	[ompi_check_pvfs2_happy="no"])
-
-    AS_IF([test "$ompi_check_pvfs2_happy" = "yes"],
-	[AS_IF([test -d "$with_pvfs2/lib"],
-		[check_pvfs2_LDFLAGS="-L$with_pvfs2/lib"
-		    $1_LDFLAGS="$check_pvfs2_LDFLAGS"
-		    LDFLAGS="$LDFLAGS $check_pvfs2_LDFLAGS"],
-		[ompi_check_pvfs2_happy="no"])
-    ],[])
-
-    # Try to find all the pvfs2 libraries
-    AS_IF([test "$ompi_check_pvfs2_happy" = "yes"],
-	[ AS_IF([test -n "$with_pvfs2_libs"]
-		[for lib in $with_pvfs2_libs ; do
-		    check_pvfs2_LIBS="$check_pvfs2_LIBS -l$lib"
-		    done])
-
-	    $1_LIBS="$check_pvfs2_LIBS"
-	    LIBS="$LIBS $check_pvfs2_LIBS"
+    OPAL_CHECK_PACKAGE([$1], [pvfs2.h], [pvfs2], [PVFS_util_resolve], [-lpthread],
```

Comment:
Does simply using `$PTHREAD_LIBS` work?

Or has OPAL_CONFIG_POSIX_THREADS already put the relevant token(s) in `$LIBS` and you don't need to specify anything at all...?